src/main/java/frc/robot/subsystems/swerve/SwerveSubsystem.java

```diff
@@ -518,6 +499,24 @@ private void updateOdometry() {
     }
   }
 
+  /** Generates a set of samples without using the async thread */
```

Comment:
Is there a reason not to reuse the results from the thread? Looks like they're not being waited on anyway, so they should be similarly up to date?

Reply:
this is mainly for the sim, where the thread doesn't run. it would also kick in if the thread fails for some reason. on the 2025 codebase the robot will log an `Alert` if this is in use instead of the thread, so that the latter case is easier to find and fix.